torch/_dynamo/output_graph.py

```diff
@@ -114,6 +118,10 @@ def __init__(
         # all current placeholder node names
         self.name_to_input = collections.OrderedDict()
 
+        # Used to avoid iterating over the linked list of graph nodes (slow)
+        # the keys contain the nodes, all values are None
+        self.nodes = collections.OrderedDict()
```

Comment:
This adds complexity.  What is the perf difference of this versus just the timestamp change?

Reply:
Yeah this was more experimental, I can probably remove this if we no longer copy the graph, `restore_checkpoint` is obviously called a lot less.